config/config.go

```diff
@@ -688,6 +819,18 @@ func (c *Config) buildClientCertPool() error {
 		}
 	}
 	c.ClientCAPool = certPool
+
+	if c.VerifyClientCertificateMetadata != nil {
```

Comment:
Instead of seeing if this property is nil, can you add a separate boolean property `VerifyClientCertificatesBasedOnProvidedMetadata` or something? I have run into issues in the past what I try to turn an obj for a feature into a boolean identifier, because then invariably, we add another property for that feature, and then things become a muddled mess.